src/FakeItEasy/Core/AssemblyExtensions.cs

```diff
@@ -14,7 +14,16 @@ public static bool ReferencesFakeItEasy(this Assembly assembly)
         {
             Guard.AgainstNull(assembly, nameof(assembly));
 
+#if FEATURE_REFLECTION_GETASSEMBLIES
             return assembly.GetReferencedAssemblies().Any(r => r.FullName == TypeCatalogue.FakeItEasyAssembly.FullName);
+#else
+            var fakeItEasyLibraryName = TypeCatalogue.FakeItEasyAssembly.GetName().Name;
+            var context = Microsoft.Extensions.DependencyModel.DependencyContext.Default;
+
+            var runtimeLib = context.RuntimeLibraries
+                .SingleOrDefault(library => string.Equals(library.Name, assembly.Name(), System.StringComparison.Ordinal));

```

Comment:
@jeremymeng, do you know why it's okay to equate a library name with an assembly name?
It's not that I believe it will stop working (it's probably the case that there will be name agreement in all but pathological cases), but I'm just wondering about the justification, aside from perhaps observation that it works.
The lack of documentation foor these things is killing me.